extension/storage/dbstorage/README.md

```diff
@@ -0,0 +1,38 @@
+# Database Storage
+
+> :construction: This extension is in alpha. Configuration and functionality are subject to change.
+
+The Database Storage extension can persist state to a relational database. 
+
+The extension requires read and write access to a database table.
+
+`driver`: the name of the database driver to use. By default, the storage client supports "sqlite3" and "pgx".
+
+Implementors can add additional driver support by importing SQL drivers into the program.
```

Comment:
How would it work concretely? I can't think of a clean way of doing that using the builder, and not using it means having to manage more than just an import statement. But perhaps I'm missing something?

Reply:
From what I get, folks building their own distros can add imports of additional drivers and import them using the underscore import, just like this SO thread explains: https://stackoverflow.com/questions/21220077/what-does-an-underscore-in-front-of-an-import-statement-mean
As long the init() function of those side-effect imports registers the driver, it should be fine.

Reply:
Right, but this import has to be in a source file that is included in the distribution. So, either a downstream distribution creates a whole new extension just for the import statement, or they have to maintain the source code for their distribution (main.go essentially) because of that import line. Or perhaps there's a better solution I'm not seeing?

Reply:
Yes, I guess they'd add that to their `main.go`? I just don't know any better, I'm sorry.

Reply:
Right -- they might just not have a main.go yet. I'm OK with approving and merging this as is, but would still be good to have an end-to-end example for users interested in extending with their SQL drivers.

Reply:
Would you please file an issue, sorry - especially interested in where such a sample would live?

Reply:
Done: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7157